spinalcordtoolbox/utils/shell.py

```diff
@@ -40,13 +40,14 @@ def display_open(file, message="Done! To view results"):
 SUPPORTED_VIEWERS = ['fsleyes', 'fslview_deprecated', 'fslview', 'itk-snap', 'itksnap']
 
 
-def display_viewer_syntax(files, verbose, colormaps=[], minmax=[], opacities=[], mode=''):
+def display_viewer_syntax(files, verbose, colormaps=[], im_types=[], minmax=[], opacities=[], mode=''):
     """
     Print the syntax to open a viewer and display images for QC. To use default values, enter empty string: ''
     Parameters
     ----------
     files [list:string]: list of NIFTI file names
     colormaps [list:string]: list of colormaps associated with each file. Available colour maps: see dict_fsleyes
+    im_types [list:string]: list of image type associated with each file. Available types: see IMTYPE_COLORMAPS
```

Comment:
The biggest design question I have is whether we want to:

1. Support both `colormaps` _and_ `im_types`, or 
2. Replace `colormaps` with `im_types` entirely

The reason I suggest "replacement" is because: 

* Someone might mistakenly supply both `colormaps` _and_ `im_types`, and get some unexpected/confusing results.
* The old `colormaps` behavior still has [several criticisms](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4030#issue-1574970648). So, is it even worth keeping around?

If we choose option 2. and get rid of `colormaps`, it does mean we would have to search the codebase for each call to `display_viewer_syntax`, and make the replacements (e.g. `colormaps=['grey', 'red']` would become `im_types=['anat', 'seg']`.

This would be a bit of work, but at least it would be a very straightforward swap? :slightly_smiling_face:

Reply:
> The biggest design question I have is whether we want to:
> 
> 1. Support both `colormaps` _and_ `im_types`, or
> 2. Replace `colormaps` with `im_types` entirely
> 
> The reason I suggest "replacement" is because:
> 
> * Someone might mistakenly supply both `colormaps` _and_ `im_types`, and get some unexpected/confusing results.

That was one of my concerns too

> * The old `colormaps` behavior still has [several criticisms](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4030#issue-1574970648). So, is it even worth keeping around?
> 
> If we choose option 2. and get rid of `colormaps`, it does mean we would have to search the codebase for each call to `display_viewer_syntax`, and make the replacements (e.g. `colormaps=['grey', 'red']` would become `im_types=['anat', 'seg']`).
> 
> This would be a bit of work, but at least it would be a very straightforward swap? 🙂

Maybe we could ask Julien before commiting to the task ?

Reply:
Right, to be honest I'm not sure what is the best design approach in this case. Maybe that should be discussed in a live meeting. That was not such an 'good first issue' at the end  😅

Reply:
Personally, my plan when I opened the issue was that we would replace the old `colormaps` argument. _(The work is pretty straightforward, and honestly shouldn't take much time, since many of the calls to `display_viewer_syntax` default to grayscale (and so they don't even specify `colormaps` to begin with).)_

> Maybe that should be discussed in a live meeting

This is a relatively small/unimportant change (just internal usage for this function, nothing user-facing), so I don't know if this needs group input necessarily?

Reply:
> Replace `colormaps` with `im_types` entirely

Just to get the ball rolling, I've implemented this proposal in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4037/commits/b199b2f9c7c807fa0eedfafc7b7f83d2f201424e.

I'm happy to revert the changes if anyone has any major objections? But, I figured that proposing a decision by demonstrating the work involved would help us get unstuck here. :)

Reply:
Given that @mguaypaq is in favor of avoiding parallel systems (https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4037#pullrequestreview-1329079482), if there are no other objections, I think we can mark this discussion as resolved?